File: BankAccount.java
public class BankAccount {
    private long accountNumber;
    private double amount;
    private char type;
    //Método Constructor de la clase, para dos de sus atributos.
    public BankAccount(long accountNumber, char type) {
        this.accountNumber = accountNumber;
        this.type = type;
    }
    public long getAccountNumber() {
        return accountNumber;
    }

    public double getAmount() {
        return amount;
    }


    public char getType() {
        return type;
    }

    public void setType(char type) {
        this.type = type;
    }

    public void setAmount(double amount) {
        if (type=='A') setAmountA(amount);
        else if (type=='B') setAmountB(amount);
        else this.amount += amount;
    }

    public void setAmountA(double amount){
        if (this.amount + amount <= 50000) this.amount += amount;
        else System.out.println("No puede tener más de 50,000. Su cuenta es tipo A");
    }

    public void setAmountB(double amount){
        if (this.amount + amount <= 100000) this.amount += amount;
        else System.out.println("No puede tener más de 100,000. Su cuenta es tipo B");
    }

    public void drawMoney(double amount){
        if (type=='A') drawMoneyA(amount);
        else if (type=='B') drawMoneyB(amount);
        else if (type=='C') drawMoneyC(amount);
    }
    public void drawMoneyA(double amount){
        if (this.amount >= 1000 && this.amount - amount >= 1000) this.amount -= amount;
        else System.out.println("Error. Saldo insuficiente o límite alcanzado en la cuenta");
    }

    public void drawMoneyB(double amount){
        if (this.amount >= 5000 && this.amount - amount >= 5000) this.amount -= amount;
        else System.out.println("Error. Saldo insuficiente o límite alcanzado en la cuenta");
    }

    public void drawMoneyC(double amount){
        if (this.amount >= 10000 && this.amount - amount >= 10000) this.amount -= amount;
        else System.out.println("Error. Saldo insuficiente o límite alcanzado en la cuenta");
    }
}

File: BankAccount.py
class BankAccount:
    def __init__(self, account_number, account_type):
        self.account_number = account_number
        self.amount = 0
        self.type = account_type

    def get_account_number(self):
        return self.account_number

    def get_amount(self):
        return self.amount

    def get_type(self):
        return self.type

    def set_type(self, account_type):
        self.type = account_type

    def set_amount(self, amount):
        if self.type == 'A':
            self._set_amount_a(amount)
        elif self.type == 'B':
            self._set_amount_b(amount)
        else:
            self.amount += amount

    def _set_amount_a(self, amount):
        if self.amount + amount <= 50000:
            self.amount += amount
        else:
            print("No puede tener más de 50,000. Su cuenta es tipo A")

    def _set_amount_b(self, amount):
        if self.amount + amount <= 100000:
            self.amount += amount
        else:
            print("No puede tener más de 100,000. Su cuenta es tipo B")

    def draw_money(self, amount):
        if self.type == 'A':
            self._draw_money_a(amount)
        elif self.type == 'B':
            self._draw_money_b(amount)
        elif self.type == 'C':
            self._draw_money_c(amount)

    def _draw_money_a(self, amount):
        if self.amount >= 1000 and self.amount - amount >= 1000:
            self.amount -= amount
        else:
            print("Error. Saldo insuficiente o límite alcanzado en la cuenta")

    def _draw_money_b(self, amount):
        if self.amount >= 5000 and self.amount - amount >= 5000:
            self.amount -= amount
        else:
            print("Error. Saldo insuficiente o límite alcanzado en la cuenta")

    def _draw_money_c(self, amount):
        if self.amount >= 10000 and self.amount - amount >= 10000:
            self.amount -= amount
        else:
            print("Error. Saldo insuficiente o límite alcanzado en la cuenta")